### PR TITLE
Update ruby

### PR DESCRIPTION
**Pull request by noahp**
Old ruby doesn't work on my ubuntu machine default version.

Just update. We're so out of date!

I confirmed the block diagrams render correctly still:

https://noahp-update-ruby.interrupt.pages.dev/blog/device-firmware-update-cookbook#dfu-code-should-be-updatable

image

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
-blockdiag
-Pillow==9.5.0
+# Install a newer version of blockdiag, that's compatible with modern pillow
+git+https://github.com/blockdiag/blockdiag.git@refs/pull/179/head
```
Comment (gminn) [inline requirements.txt:2]: Frustrating that they haven't merged this yet

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-2.7.7
\ No newline at end of file
+3.2.2
```
Comment (gminn) [inline .ruby-version:1]: Was there a particular reason to go to 3.2.2 instead of latest (3.3.3)?

**Review by gminn — APPROVED** [review]
😮‍💨 2.7 is [already EOL](https://www.ruby-lang.org/en/downloads/branches/). thanks for upgrading!

File: .ruby-version (hunk 2 above)
Comment (noahp) [reply .ruby-version:1]: unfortunately the cloudflare builder is on 3.2.2, and updating it in `cloudflare.sh` didn't seem to work :/

File: cloudflare_build.sh [hunk 3]
```diff
@@ -1 +1,2 @@
-bundle exec jekyll build --config _config.yml${JEKYLL_BUILD_CONFIG_PARAM} && cp _redirects _site/_redirects
\ No newline at end of file
+asdf plugin update ruby
```
Comment (noahp) [inline cloudflare_build.sh:1]: revert!

File: .ruby-version (hunk 2 above)
Comment (gminn) [reply .ruby-version:1]: ah makes sense. sad

**Pull request merged**